angel_system/impls/detect_activities/swinb/swinb_detect_activities.py

```diff
@@ -0,0 +1,211 @@
+        assert len(frame_iter) == (self._sampling_rate * self._num_frames)
+        model = self.get_model()
+
+        # Form the frames into the required format for the video model
+        # Based off of the Learn swin CollateFn
+        spatial_idx = 1 # only perform uniform crop and short size jitter
+        clip_idx = -1
+
+        frames = [self.transform(f) for f in frame_iter]
+        frames = [torch.stack(frames)]
+
+        fps, target_fps = 30, 30
+        clip_size = self._sampling_rate * (self._num_frames) / target_fps * fps
+        start_end_idx = [
+            get_start_end_idx(len(x), clip_size, clip_idx=clip_idx, num_clips=1)
+            for x in frames
+        ]
+
+        # This subsamples every n (sample rate) frames
+        frames = [
+            temporal_sampling(x, s, e, self._num_frames)
+            for x, (s, e) in zip(frames, start_end_idx)
+        ]
+
+        # Crop and random short side scale jitter
+        frames = [x.permute(1, 0, 2, 3) for x in frames]
+        frames = [spatial_sampling(x, spatial_idx=spatial_idx, min_scale=224, max_scale=224, crop_size=224) for x in frames]
+        frames = torch.stack(frames)
+
+        # Move the inputs to the GPU if necessary
+        if self._model.cuda:
+            frames = frames.cuda()
+
+        # Predict!
+        with torch.no_grad():
+            preds = self._model(frames)
+
+        # Get the top predicted classes
+        post_act = torch.nn.Softmax(dim=1)
+        preds: torch.Tensor = post_act(preds) # shape: (1, num_classes)
+        top_preds = preds.topk(k=5)
+
+        # Map the predicted classes to the label names
+        # top_preds.indices is a 1xk tensor
+        pred_class_indices = top_preds.indices[0]
+
+        pred_class_names = [self._labels[int(i)] for i in pred_class_indices]
+
+        # Filter out any detections below the threshold
+        predictions = []
+        pred_values = top_preds.values[0]
+        for idx, p in enumerate(pred_class_names):
+            if (pred_values[idx] > self._det_threshold):
+                predictions.append(p)
+
+        return predictions
+
+    def get_config(self) -> dict:
+        return {
+            "use_cuda": self._use_cuda,
+            "cuda_device": self._cuda_device,
+            "num_classes": self._num_classes,
+            "det_threshold": self._det_threshold,
+            "checkpoint_path": self._checkpoint_path,
+            "num_frames": self._num_frames,
+            "sampling_rate": self._sampling_rate,
+            "labels_file": self._labels_file,
+        }
+
+    @classmethod
+    def is_usable(cls) -> bool:
+        # check for optional dependencies
+        torch_spec = importlib.util.find_spec('torch')
+        torchvision_spec = importlib.util.find_spec('torchvision')
+        if (
+            torch_spec is not None and
+            torchvision_spec is not None
+        ):
+            return True
+        else:
+            return False
```

Comment:
Since the context of this package is that it requires torch/torchvision, this could just simply return True, which means we don't need to implement it as the parent definition simply returns True.